src/grid/interpolate.jl

```diff
@@ -98,3 +99,72 @@ linear interpolation of data(x, y)
     gx = (g0 * dy1 + g1 * dy0) / (dx0 + dx1) / (dy0 + dy1)
     return gx
 end
+
+
+"""
+testInterpolation1D(func, grid1, grid2)
+
+Test interpolation on grid1 with function func, compare results with func(grid2).
+Return max deviation and std of deviation.
+
+"""
+function testInterpolation1D(func, grid1, grid2)
+    data1 = [func(grid1[i]) for i in 1:(grid1.size)]
+
+    data2 = [func(grid2[i]) for i in 1:(grid2.size)]
+    data2_int = [linear1D(data1, grid1, grid2[i]) for i in 1:(grid2.size)]
+
+    d_max = maximum( abs.( data2-data2_int ))
+    d_std = std(data2-data2_int)
+    return d_max, d_std
+end
+
+function testInterpolation1D_rel(func, grid1, grid2)
+    data1 = [func(grid1[i]) for i in 1:(grid1.size)]
+
+    data2 = [func(grid2[i]) for i in 1:(grid2.size)]
+    data2_int = [linear1D(data1, grid1, grid2[i]) for i in 1:(grid2.size)]
+
+    d_rel = abs.((data2-data2_int) ./ data2)
+    d_max = maximum(d_rel)
+    d_std = std(d_rel)
+    return d_max, d_std
+end
+
+function optimizeUniLog(generator, para, MN, func)
```

Comment:
Use the standard format for the documentation

"""
function optimizeUniLog(generator, para, MN, func)

return an automatically optimized UniLog grid....

# Arguments:

- `generator`: function `generator(para, M, N)` to generate a UniLog grid
-  ...
"""